Source/Core/Core/HW/WiimoteEmu/Extension/BalanceBoard.cpp

```diff
@@ -0,0 +1,213 @@
+                        // Bottom right 0kg
+                        u8(WEIGHT_0_KG >> 8), u8(WEIGHT_0_KG & 0xFF),
+                        // Top left 0kg
+                        u8(WEIGHT_0_KG >> 8), u8(WEIGHT_0_KG & 0xFF),
+                        // Bottom left 0kg
+                        u8(WEIGHT_0_KG >> 8), u8(WEIGHT_0_KG & 0xFF),
+                        // Top right 17kg
+                        u8(WEIGHT_17_KG >> 8), u8(WEIGHT_17_KG & 0xFF),
+                        // Bottom right 17kg
+                        u8(WEIGHT_17_KG >> 8), u8(WEIGHT_17_KG & 0xFF)}};
+  m_reg.calibration2 = {{// Top left 17kg
+                         u8(WEIGHT_17_KG >> 8), u8(WEIGHT_17_KG & 0xFF),
+                         // Bottom left 17kg
+                         u8(WEIGHT_17_KG >> 8), u8(WEIGHT_17_KG & 0xFF),
+                         // Top right 34kg
+                         u8(WEIGHT_34_KG >> 8), u8(WEIGHT_34_KG & 0xFF),
+                         // Bottom right 34kg
+                         u8(WEIGHT_34_KG >> 8), u8(WEIGHT_34_KG & 0xFF),
+                         // Top left 34kg
+                         u8(WEIGHT_34_KG >> 8), u8(WEIGHT_34_KG & 0xFF),
+                         // Bottom left 34kg
+                         u8(WEIGHT_34_KG >> 8), u8(WEIGHT_34_KG & 0xFF),
+                         // Checksum - computed later
+                         0xff, 0xff, 0xff, 0xff}};
+  m_reg.bboard_unknown = {{// 2 used unknown bytes
+                           0x19, 0x01}};
+
+  ComputeCalibrationChecksum();
+}
+
+ControllerEmu::ControlGroup* BalanceBoard::GetGroup(BalanceBoardGroup group)
+{
+  switch (group)
+  {
+  case BalanceBoardGroup::Balance:
+    return m_balance;
+  case BalanceBoardGroup::Weight:
+    return m_weight;
+  default:
+    assert(false);
+    return nullptr;
+  }
+}
+
+void BalanceBoard::DoState(PointerWrap& p)
+{
+  EncryptedExtension::DoState(p);
+}
+
+void BalanceBoard::LoadDefaults(const ControllerInterface& ciface)
+{
+  // Stick
+  m_balance->SetControlExpression(0, "I");  // up
+  m_balance->SetControlExpression(1, "K");  // down
+  m_balance->SetControlExpression(2, "J");  // left
+  m_balance->SetControlExpression(3, "L");  // right
+
+  // Because our defaults use keyboard input, set calibration shape to a square.
+  m_balance->SetCalibrationFromGate(ControllerEmu::SquareStickGate(.5));
+}
+
+u16 BalanceBoard::ConvertToSensorWeight(double weightInKilos)
+{
+  // Note: this is the weight on a single sensor, so these ranges make more sense
+  // (if all sensors read 34 kilos, then the overall weight would be 136 kilos or 300 pounds...)
+  if (weightInKilos < 17)
+  {
+    return WEIGHT_0_KG + (LOW_WEIGHT_DELTA * weightInKilos / 17);
+  }
+  else
+  {
+    return WEIGHT_17_KG + (HIGH_WEIGHT_DELTA * (weightInKilos - 17) / 17);
+  }
+}
+
+void BalanceBoard::ComputeCalibrationChecksum()
+{
+  u8 data[0x1c];
+  // Ugh.  std::copy_n or something would be better, but it doesn't work well with the offsets.
```

Comment:
This should probably be removed if there's no plan to use copy or copy_n, given it doesn't really add anything to the code itself.